api/src/main/java/org/apache/iceberg/UpdateSnapshotReference.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Map;
+
+/**
+ * API for updating snapshot reference.
+ * <p>
+ * Apply returns the updated snapshot reference as a {@link SnapshotReference} for validation.
+ * <p>
+ * When committing, these changes will be applied to the current table metadata. Commit conflicts
+ * will be resolved by applying the pending changes to the new table metadata.
+ */
+public interface UpdateSnapshotReference extends PendingUpdate<Map<String, SnapshotReference>> {
+
+  /**
+   * remove snapshotReference.
+   *
+   * @param name name of snapshot reference
+   * @return this
+   * @throws IllegalArgumentException If there is no such snapshot reference named name
+   */
+  UpdateSnapshotReference removeReference(String name);
+
+  /**
+   * Update minSnapshotsToKeep of snapshotReference what will be search by referenceName and snapshotReferenceType.
+   *
+   * @param ageMs       new maxSnapshotAgeMs for snapshot reference. If null will not update.
+   * @param numToKeep   new minSnapshotsToKeep for snapshot reference. If null will not update.
+   * @param name        name of snapshot reference what will be update
+   * @return this
+   */
+
+  UpdateSnapshotReference setBranchRetention(String name, Long ageMs, Integer numToKeep);
```

Comment:
I think we should use primitive values so that people must specify a value. Having a method call like `setBranchRetention("test", null, null)` is unfriendly to readers of the code.

But I understand we want to have the flexibility for users to use defaults. We can potentially introduce 2 different methods for these two settings separately, like `setBranchSnapshotLifetime(String name, long ageMs)` and `setMinSnapshotsInBranch(String name, int num)`, but maybe its a bit overkill.

Reply:
I used code several method like setBranchSnapshotLifetime and setMinSnapshotsInBranch. But i also think its a bit overkill, if it's necessary i will add these method.